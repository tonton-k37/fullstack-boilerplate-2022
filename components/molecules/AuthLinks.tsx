import { Button as ChakraButton } from "@chakra-ui/react";
import { useSession, signOut } from "next-auth/react";
import NextLink from "next/link";
import { memo } from "react";
import { Button } from "@components/atoms";

const AuthLinks = memo(() => {
  const { data: session } = useSession();

  const buttonProps = {
    onClick: () => signOut(),
  };

  if (session) {
    return <Button buttonProps={buttonProps} text="sign out" />;
  }

  return (
    <>
      <NextLink href="./signup" passHref>
        <ChakraButton as="a">sign up</ChakraButton>
      </NextLink>
      <NextLink href="./signin" passHref>
        <ChakraButton as="a">sign in</ChakraButton>
      </NextLink>
    </>
  );
});

export { AuthLinks };
