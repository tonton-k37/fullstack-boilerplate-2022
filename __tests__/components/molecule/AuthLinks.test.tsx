import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { AuthLinks } from "@components/molecules";

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual("next-auth/react");
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
    user: { username: "admin" },
  };
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return { data: mockSession, status: "authenticated" };
    }),
  };
});

describe("AuthLinks Component", () => {
  it("Show Log Out When no session", async () => {
    const { container } = render(<AuthLinks />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText("sign out")).toBeInTheDocument();
  });
});
