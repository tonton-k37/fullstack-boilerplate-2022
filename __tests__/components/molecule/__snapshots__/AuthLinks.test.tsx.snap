// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLinks Component Show Log Out When no session 1`] = `
<div>
  <button
    class="chakra-button css-4xx2wk"
    type="button"
  >
    sign out
  </button>
</div>
`;
